### ISSUE TITLE
Q: interactive plotting?

### ISSUE THREAD

**Issue by kingjr** [nilearn/nilearn#969]
Hi all,

Sorry if this is not the best place to ask this.

I am looking for an interactive MRI viewer in Python to semi-automatically locate intracranial electrodes based on a priori knowledge (respective locations, on cortical surface etc), and manual inputs.

Such viewer is standard in MRI packages:  3 views (x, y, z axes), similarly to `plotting.plot_anat()` but with some interaction: if you click on one voxel, the 3 views update their slice. 

I started a [gist](https://gist.github.com/kingjr/bb9e732fe8df6a8cd571), but it doesn't include all the nilearn viewing options (e.g. the xyz scales are in voxel units, not in mm)

I have tried to directly use nilearn plotting functions, but they seem to be strictly static, and I've encountered difficulties in finding a way to update an overlay, because I got lost in the display and slicer classes.

So in short, is the update of a MRI plot possible ? e.g.
```
display = plotting.plot_anat(fname, display_mode='x', cut_coords=[0], axes=ax)
plt.show()
display.update(cut_coords[10])
plt.show()
```

or even something like

```
brain_slice, extent = get_slice(anat, mode='x', cut_coords[0])
plt.imshow(brain, extent=extent)
```

Else, would it be conceivable? I'm happy to make a PR, but I'll need directions..

Thanks a lot for your help!

**Comment by GaelVaroquaux** [comment]
For now interactice plotting is out of scope for nilearn. We are still
looking to extend the processing functionality of nilearn before
considering more visualization.

In addition, interactive data visualization and non-interactive tend to
require very different compromises. Nilearn's compromises are toward
non-interactive. The update of the data is not possible. Typically,
different underlying plotting libraries are good at interactive
visualization or non-interactive. Matplotlib, on which we rely for
nilearn, is currently not very good at interactive visualization.

In the nilearn sandbox
(https://github.com/nilearn/nilearn_sandbox/tree/master/nilearn_sandbox)
we have a prototype using a javascript visualizer for interactive
visualization in the browser.

**Comment by kingjr** [comment]
Ok, thanks for the quick reply

**Comment by eickenberg** [comment]
maybe https://github.com/AlexandreAbraham/pynax/
can be of use?

**Comment by kingjr** [comment]
> maybe https://github.com/AlexandreAbraham/pynax/ can be of use?

Looks like it, thanks!
